### DISCUSSION TITLE
More experiences with the vendure multivendor support

### DISCUSSION THREAD

**Discussion by tlodge** [vendurehq/vendure#2166] [Q&A]
I'm slowly getting there I think.  I have been updating the remix storefront to get it working with v2.  I made some changes to the `providers/orders/order.ts` to update the gql, as suggested [here](https://github.com/vendure-ecommerce/storefront-remix-starter/issues/40).   Plus I changed setOrderShippingMethod to:

```
export function setOrderShippingMethod(
  shippingMethodId: [string],
  options: QueryOptions,
) {
  return sdk.setOrderShippingMethod({shippingMethodId}, options);
}
```

I also had to update the gql in `providers/customer/customer.ts' to work with the new fulfillmentLines schema:

```
gql`
  query activeCustomerOrderList($orderListOptions: OrderListOptions) {
    activeCustomer {
      orders(options: $orderListOptions) {
        totalItems
        items {
          code
          state
          orderPlacedAt
          currencyCode
          subTotal
          subTotalWithTax
          total
          totalWithTax
          shippingWithTax
          shippingLines {
            priceWithTax
          }
          taxSummary {
            taxBase
            taxTotal
          }
          discounts {
            amountWithTax
          }
          lines {
            quantity
            discountedLinePriceWithTax
            discountedUnitPriceWithTax
            fulfillmentLines{
              fulfillment{
                  updatedAt
                  state
              }
            }
            featuredAsset {
              name
              source
              preview
            }
            productVariant {
              name
              sku
              currencyCode
              priceWithTax
              product {
                slug
              }
            }
          }
        }
      }
    }
  }
`;
```

and I changed `routes/api/active-order.tsx` line 94(ish) to:

```
 const result = await setOrderShippingMethod([shippingMethodId], {
          request,
        });
```

I then updated `codegen.yml` to point to my local instance of vendure and then ran

```
npm run generate
```
and all worked well.  However, when I then select two items to buy in remix, the multivendor plugin fails.  It looks like it's because the (non-bobs-parts) item that I am trying to purchase doesn't has a null seller channelId, so the function `assignShippingLineToOrderLines` in `mv-shipping-line-assignment-strategy.ts` fails at the line:

```
return order.lines.filter(line => {
                    return  idsAreEqual(line.sellerChannelId, sellerChannel.id)
});
```

since line.sellerChannelId is null for the non-bobs-parts item.

I'm assuming that this is failing because the default products in vendure aren't assigned to a seller?

Thanks again!

**Comment by dylviz** [comment]
What happens if you create 2 new sellers then create new products from the seller's accounts?

**Reply by tlodge** [reply]
Hi @dylviz - thanks.  I think that would work fine (though haven't checked yet).  I think the issue may be that when an order is created for a product that belongs (only) to the default channel, the order line channelId is not set (rather than being set to the default channelId).  For now I've just fixed it in the plugin, to do a null check and set the line.sellerChannelId to the default channelId if null.  I'll look into it a bit more when I get the chance. It's not really a big issue as I guess the standard case will be to start with a bunch of sellers and to add products to each.

**Reply by dylviz** [reply]
But in the multivendor case, sellerChannelID needs a non-default channel to belong to. Otherwise, you'll need to write a new strategy to process the orders, is that right?

I'm mainly curious to see that once you get past this issue, my guess is that you'll run into the same issue I have (here)[https://github.com/vendure-ecommerce/vendure/discussions/2167]. But I could be wrong

**Comment by tlodge** [comment]
Yep probably. I’m going to assume all products are associated with a seller
for now and not worry about it. I had a quick read of the issue you posted.
I’ll try and recreate and have a dig around too. It probably won’t be until
the end of the week though.

**Reply by dylviz** [reply]
Ya no worries, having these discussions just helps me better understand the platform. Though I'm sure updates are on the way to fix a lot of this, my curiosity gets the better of me lol I'm gonna mess around with some front end for a bit
